tests/poller.spec.js

```diff
@@ -168,55 +168,29 @@ describe ('Poller', function () {
 
 	});
 
-	it ('Should throw from getData when fetch has received an HTTP error', function (done) {
```

Comment:
No, it should not do this. It should return the cached data successful fetch

Reply:
Yeah, I'm having trouble remembering why we did that in #91. It might make more sense to log that it's returning stale data?

Reply:
@alexmuller and I were just discussing this. I think we're in agreement that this shouldn't throw (although we'd like to work out why we did this). We identified four possible states that the poller can be in and think it might be better to log and return the relevant data in each case. Thoughts, @kavanagh?

Maybe we could repurpose this PR to cater for all these cases as you've already done the work to return stale data?

## Case 1: No data has been fetched yet

1. We send an **info**-level log that we’re serving default data
2. We return **default data**

## Case 2: All fetches errored, there was never a successful fetch

1. We send an **error**-level log that we’re serving default data and cannot get fresh data
2. We return **default data**

## Case 3: A fetch in the past was successful, the latest fetch errored

1. We send a **warning**-level log that we’re serving stale data
2. We return **stale data**

## Case 4: The latest fetch was successful, getData returns fresh data

1. No log needed
2. We return **fresh data**

Reply:
I like this. It's a light touch improvement that's at least 200% better than what we have.